src/parsley.cpp

```diff
@@ -3483,17 +3483,19 @@ AtomicInfo *APar_reverseDNS_atom_Init(const char *rDNS_atom_name,
                               ilst_atom->AtomicLevel + 1,
                               last_iTunes_list_descriptor);
 
+    uint32_t domain_len = strlen(rDNS_domain);
     short rDNS_mean_atom = APar_InterjectNewAtom("mean",
                                                  CHILD_ATOM,
                                                  VERSIONED_ATOM,
-                                                 12,
+                                                 domain_len,
```

Comment:
This should be domain_len + 1 to deal with an edge case where a null terminator will missing from internal atom value when domain_len == 16. See:

https://github.com/wez/atomicparsley/blob/d813aa6e0304ed3ab6d92f1ae96cd52b586181ec/src/parsley.cpp#L2389-L2394

The same would apply to the "name" atom, where name_len should be name_len + 1  at:

https://github.com/wez/atomicparsley/blob/d813aa6e0304ed3ab6d92f1ae96cd52b586181ec/src/parsley.cpp#L3503-L3510

Reply:
It seems that the edge case is remedied without these changes if the atom length is initialised to 12 instead of 16 (below)